testnet/stacks-node/src/tests/neon_integrations.rs

```diff
@@ -5820,3 +5819,313 @@ fn atlas_stress_integration_test() {
+
+    // Publish the contract so we can use it.
+    submit_tx(
+        &http_origin,
+        &make_contract_publish(
+            &creator_sk,
+            0,
+            1100000,
+            "increment-contract",
+            &giant_contract,
+        ),
+    );
+
+    // Wait to make sure the contract is published.
+    run_until_burnchain_height(
+        &mut btc_regtest_controller,
+        &blocks_processed,
+        &burnchain_height,
+        215,
+    );
+
+    // Check that we have defined the contract.
+    let increment_contract_defines = select_transactions_where(
+        &test_observer::get_blocks(),
+        |transaction| match &transaction.payload {
+            TransactionPayload::SmartContract(contract) => {
+                contract.name == ContractName::try_from("increment-contract").unwrap()
+            }
+            _ => false,
+        },
+    );
+    assert_eq!(increment_contract_defines.len(), 1);
+
+    // Alice calls the contract and should succeed, because we have not lowered the block limit
+    // yet.
+    submit_tx(
+        &http_origin,
+        &make_contract_call(
+            &alice_sk,
+            0,
+            1000,
+            &creator_addr.into(),
+            "increment-contract",
+            "increment-many",
+            &[Value::UInt(0 as u128)],
+        ),
+    );
+
+    // Wait for the contract call to process.
+    run_until_burnchain_height(
+        &mut btc_regtest_controller,
+        &blocks_processed,
+        &burnchain_height,
+        220,
+    );
+
+    // Check that we have processed the contract successfully, by checking that the contract call
+    // is in the block record.
+    let increment_calls_alice = select_transactions_where(
+        &test_observer::get_blocks(),
+        |transaction| match &transaction.payload {
+            TransactionPayload::ContractCall(contract) => {
+                contract.contract_name == ContractName::try_from("increment-contract").unwrap()
+            }
+            _ => false,
+        },
+    );
+    assert_eq!(increment_calls_alice.len(), 1);
+
+    // Clear the observer so we can look for Bob's transaction.
+    test_observer::clear();
+
+    // The cost contract was switched at height 220. So, now we expect Bob's call to fail.
+    run_until_burnchain_height(
+        &mut btc_regtest_controller,
+        &blocks_processed,
+        &burnchain_height,
+        230,
+    );
+    submit_tx(
+        &http_origin,
+        &make_contract_call(
+            &bob_sk,
+            0,
+            1000,
+            &creator_addr.into(),
+            "increment-contract",
+            "increment-many",
+            &[Value::UInt(0 as u128)],
+        ),
+    );
+
+    // Wait for the contract to finish.
+    run_until_burnchain_height(
+        &mut btc_regtest_controller,
+        &blocks_processed,
+        &burnchain_height,
+        235,
+    );
+
+    // Bob's calls didn't work because he called after the block limit was lowered.
```

Comment:
Is there a way to query the miner for transactions that got rejected?  Then you could verify that the miner tried to apply Bob's transaction, but failed to do so.

Reply:
No, I don't believe that there's a way to do this.

Reply:
I think this is something that may be worthwhile to add to our integration testing framework, but it's pretty unrelated to this PR -- many of our integration tests already do this kind of "assumed transaction rejection" testing (all of the overflow tests).

Reply:
If we want to continue discussing this, let's do it on the PR, not the commit!